### PR TITLE
Station hierarchy guidance

### PR DESCRIPTION
**Pull request by gknisely**
This is a lot of updates because it is a combination of updates from myself and @dgearhart.   

The transit pbf data is now serialized in the following order.  
Egress1_for_Station_A
Egress2_for_Station_A
Station_A
Platform1_for_Station_A
Platform2_for_Station_A
Egress_for_Station_B
Station_B
Platform_for_Station_B
. . . and so on

The data is now saved in the transit tiles (level 3) as follows:
Tiles will look like the following with N egresses and N platforms.

From the road network...
```text
                                                          level 3
                \-----------------------------------------------------------------------------------------\
                \                                                     rail /              rail /          \
        tc      \               ec                 pc                 bus                 bus             \
osm------|------\--->egress-----|----->station-----|----->platform-----|----->platform-----|-----> . . .  \
node            \     node              node                node                node                      \
                \                       /   \                                                             \
                \                 ec - /     \ - pc       rail /                                          \
        tc      \                     /       \           bus                                             \
osm------|------\-------------->N egress    N platform-----|-----> . . .                                  \
                \                 nodes       nodes                                                       \
                \-----------------------------------------------------------------------------------------\
```
To the road network...
```text
                                                          level 3
                \-----------------------------------------------------------------------------------------\
                \                                                     rail /              rail /          \
        tc      \               ec                 pc                 bus                 bus             \
osm<------|-----\---egress<-----|-----station<-----|-----platform<-----|-----platform<-----|----- . . .   \
node            \    node              node                node                node                       \
                \                     /   \                                                               \
                \               ec - /     \ - pc         rail /                                          \
        tc      \                   /       \             bus                                             \
osm<------|-----\--------------N egress    N platform<-----|----- . . .                                   \
                \                 nodes       nodes                                                       \
                \-----------------------------------------------------------------------------------------\
```

tc:   osm and egress nodes are connected by transitconnections.
ec:  egress and stations are connected by egressconnections.
pc:  stations and platforms are connected by platformconnections

The new logic has been RAD tested using NYC, SF, PHI, and Rome data by myself and @dgearhart

### CODE REVIEW

File: src/mjolnir/graphtilebuilder.cc [hunk 1]
```diff
@@ -799,7 +799,8 @@ std::array<std::vector<GraphId>, kBinCount> GraphTileBuilder::BinEdges(const Gra
   const auto* start_edge = tile->directededge(0);
   for(const DirectedEdge* edge = start_edge; edge < start_edge + tile->header()->directededgecount(); ++edge) {
     //dont bin these
-    if(edge->is_shortcut() || edge->trans_up() || edge->trans_down() || edge->use() == Use::kTransitConnection)
+    if(edge->is_shortcut() || edge->trans_up() || edge->trans_down() || edge->use() == Use::kTransitConnection ||
```
Comment (dnesbitt61) [inline src/mjolnir/graphtilebuilder.cc]: could use new IsTransition member of DirectedEdge (in place of trans_up || trans_down)

**Comment by kevinkreiser** [comment]
that ascii art is rad! :+1: for that!

File: src/mjolnir/graphtilebuilder.cc (hunk 1 above)
Comment (gknisely) [inline src/mjolnir/graphtilebuilder.cc]: Done.  Thanks.  Did not know that existed.

File: src/thor/trippathbuilder.cc [hunk 2]
```diff
@@ -711,57 +715,131 @@ TripPath TripPathBuilder::Build(
...
+
+            // Set station name if requested
+            if (controller.attributes.at(kNodeTransitPlatformInfoStationName) && transit_station->name_offset())
+              transit_platform_info->set_station_name(endtile->GetName(transit_station->name_offset()));
+
+            //only one de to station exists.  we are done.
+            break;
+          }
+
+        }
```
Comment (dnesbitt61) [inline src/thor/trippathbuilder.cc]: consider separate method?

Comment (kevinkreiser) [inline src/thor/trippathbuilder.cc]: consider maybe like 10 different methods! its like 1400 lines of code!

Comment (gknisely) [inline src/thor/trippathbuilder.cc]: agree @kevinkreiser feather it in please.

Comment (kevinkreiser) [inline src/thor/trippathbuilder.cc]: commencing operation :dove: tail

**Comment by dnesbitt61** [comment]
:shipit:

File: src/thor/trippathbuilder.cc (hunk 2 above)
Comment (gknisely) [inline src/thor/trippathbuilder.cc]: @dnesbitt61 moved to a function.  Please review

**Pull request merged**